src/Tokenomia/Vesting/PrivateSale.hs

```diff
@@ -0,0 +1,235 @@
+{-# LANGUAGE BlockArguments #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ImportQualifiedPost #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Tokenomia.Vesting.PrivateSale (verifyPrivateSale) where
+
+import Ledger qualified
+import Ledger.Value qualified as Value
+import Prelude
+
+import Control.Monad
+import Control.Monad.Reader
+import GHC.Generics (Generic)
+
+import Data.Hex (unhex)
+import Data.String (fromString)
+
+import Control.Lens
+import Control.Monad.Except
+import Data.Aeson (FromJSON, eitherDecodeFileStrict)
+import Data.Bifunctor (first)
+import Data.Foldable (find)
+import Data.Text (Text, pack, unpack)
+import Money qualified as M
+import System.IO (FilePath, getLine, putStrLn)
+
+import Blockfrost.Client qualified as B
```

Comment:
Generally we avoid importing modules under the same qualified name and also using single-letter qualified modules names. You should look at our styleguide for how imports should be organized and adjust this